<?xml version="1.0"?>
<!--
SPDX-FileCopyrightText: 2023 Idiap Research Institute <contact@idiap.ch>

SPDX-FileContributor: Jeremy Maceiras  <jeremy.maceiras@idiap.ch>

SPDX-License-Identifier: GPL-3.0-only
-->
<robot name="iiwa" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="grey">
    <color rgba="0.85 0.85 0.85 0.5"/>
  </material>
  <!--Import the lbr iiwa macro -->
  <xacro:include filename="$(find iiwa_description)/urdf/iiwa.urdf.xacro"/>
  
  <xacro:arg name="origin_xyz" default="0 0 0"/>
  <xacro:arg name="origin_rpy" default="0 0 0"/>
  
  <!-- Fix to world just for testing -->
  <link name="world"/>
  
  <!--iiwa-->
  <xacro:iiwa prefix="iiwa_" parent="world">
    <origin xyz="$(arg origin_xyz)" rpy="$(arg origin_rpy)" />
  </xacro:iiwa>
  
  <link name="pen_link">
  </link>

  <joint name="pen_joint" type="fixed">
    <origin rpy="0 1.5708 0" xyz="0.15 0 0.12"/>
    <parent link="iiwa_tool0"/>
    <child link="pen_link"/>
  </joint>

  <link name="drawing_frame_link">
    <visual>
      <geometry>
        <mesh
          filename="package://ergodic_sketching_ros/descriptions/meshes/drawing_paper.stl"/>
      </geometry>
      <material name="grey"/>
    </visual>
  </link>

  <joint name="drawing_frame_joint" type="fixed">
    <parent link="iiwa_link_0"/>
    <child link="drawing_frame_link"/>

    <origin rpy="$(arg drawing_frame_rpy)" xyz="$(arg drawing_frame_xyz)"/>
  </joint>
</robot>
